go/indexer/services/l1/confirmed_headers.go

```diff
@@ -177,34 +187,35 @@ func (f *ConfirmedHeaderSelector) NewHead(
 		endHeight = startHeight + f.cfg.MaxBatchSize - 1
 	}
 
-	nHeaders := endHeight - startHeight + 1
+	nHeaders := int(endHeight - startHeight + 1)
 	if nHeaders > 1 {
-		logger.Info("Loading block batch ",
+		logger.Info("Loading blocks",
 			"startHeight", startHeight, "endHeight", endHeight)
 	}
 
-	headers := make([]*NewHeader, nHeaders)
-	var wg sync.WaitGroup
-	for i := uint64(0); i < nHeaders; i++ {
-		wg.Add(1)
-		go func(ii uint64) {
-			defer wg.Done()
+	headers := make([]*NewHeader, 0)
+	height := startHeight
+	left := nHeaders - len(headers)
+	for left > 0 {
+		count := DefaultMaxBatchSize
+		if count > left {
+			count = left
+		}
 
-			ctxt, cancel := context.WithTimeout(ctx, DefaultConnectionTimeout)
-			defer cancel()
+		logger.Info("Loading block batch",
+			"height", height, "count", count)
 
-			height := startHeight + ii
-			bigHeight := new(big.Int).SetUint64(height)
-			header, err := HeaderByNumber(ctxt, client, bigHeight)
-			if err != nil {
-				log.Error("Unable to load block ", "block", height, "err", err)
-				return
-			}
+		ctxt, cancel := context.WithTimeout(ctx, DefaultConnectionTimeout)
+		fetched, err := HeadersByRange(ctxt, client, height, count)
```

Comment:
Thoughts on not passing through `count` and just using the global hardcoded value inside of `HeadersByRange`?

Reply:
Ah its by range so passing in `count` makes sense